docs/src/search.md

```diff
@@ -107,6 +111,33 @@ an ANN search means that using an index often involves a trade-off between recal
 See the [IVF_PQ index](./concepts/index_ivfpq.md) for a deeper description of how `IVF_PQ`
 indexes work in LanceDB.
 
+## Binary vector
+
+LanceDB supports binary vectors as a data type, and has the ability to search binary vectors with hamming distance. The binary vectors are stored as uint8 arrays:
+
+=== "Python"
+
+    ```python
+    import lancedb
+    import numpy as np
+
+    db = lancedb.connect("data/sample-lancedb")
+
+    data = [
+        {
+            "id": i,
+            "vector": np.random.randint(0, 256, size=128),
+        }
+        for i in range(1024)
+    ]
+
+    tbl = db.create_table("my_binary_vectors", data=data)
+
+    query = np.random.randint(0, 256, size=128)
+    results = tbl.search(query).to_arrow()
+    ```
```

Comment:
Going forward, I think our preference is to put these tests in `python/tests/docs` and reference them with the snip operator `--8<--`. For example, these Python examples:

https://lancedb.github.io/lancedb/basic/#connect-to-a-database

Are here: 

https://github.com/lancedb/lancedb/blob/59b57055e78c723b84793688eeb108105026df37/docs/src/basic.md?plain=1#L137-L142

And the actual code is here:

https://github.com/lancedb/lancedb/blob/59b57055e78c723b84793688eeb108105026df37/python/python/tests/docs/test_basic.py#L15-L19

If we use these, all the examples are well tested.

Reply:
added